aidial_adapter_openai/utils/chat_completion_response.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Iterable, Literal, Self
+
+from aidial_sdk.utils.merge_chunks import merge_chat_completion_chunks
+from pydantic import BaseModel
+
+
+class ChatCompletionResponse(BaseModel):
+    message_key: Literal["delta", "message"]
+    resp: dict = {}
+
+    @property
+    def usage(self) -> Any | None:
+        return self.resp.get("usage")
+
+    @property
+    def is_empty(self) -> bool:
+        return self.resp == {}
+
+    @property
+    def finish_reasons(self) -> Iterable[Any]:
+        for choice in self.resp.get("choices") or []:
+            if (reason := choice.get("finish_reason")) is not None:
+                yield reason
+
+    @property
+    def has_finish_reason(self) -> bool:
+        return len(list(self.finish_reasons)) > 0
+
+    @property
+    def messages(self) -> Iterable[Any]:
```

Comment:
Here I think typing won't be so intelligent, so we would need to extend our generic with two types: both for response, and for choice